src/thunderbots/software/ai/navigator/util.h

```diff
@@ -18,3 +18,17 @@
  */
 double calculateTransitionSpeedBetweenSegments(const Point &p1, const Point &p2,
                                                const Point &p3, double final_speed);
+
+/**
+ * Calculates how much a point is trespassing in another point's space.
+ *
+ * Returns the distance between the given point and the threshold point if it is within
```

Comment:
The return is slightly backwards. We want to return how far inside the "trespass boundary" the point is from the other, so the trespass value is 0 outside the boundary, and increases as the point approaches the other. Should be an easy fix.

Reply:
I see! Here is my current interpretation for what the trespass value/distance is:
![untitled (5)](https://user-images.githubusercontent.com/38922275/57975314-2dedac80-797b-11e9-9e02-bd1b5fe18fbc.png)
Is this understanding correct?

As well, I for the life of me couldn't quite figure out why `dist(p1, p2)` returns an "undefined reference" error in line 14 of `util.cpp` so I used `(p1 - p2).len()` instead for now.

Reply:
Yes your understanding is correct (nice diagram btw).